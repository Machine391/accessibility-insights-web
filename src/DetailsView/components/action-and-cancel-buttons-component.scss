// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

.action-and-cancel-buttons-component {
    display: flex;
    justify-content: flex-end;

    .action-cancel-button-col + .action-cancel-button-col {
        margin-left: 8px;
    }

    .cancel-button {
        font-size: 14px;
        font-weight: 600;
        height: 32px;
        border: 1px solid $neutral-60;
        color: $neutral-60;
        border-radius: 2px;

        &:hover {
            border: 1px solid $neutral-60;
            font-weight: 600;
            color: $neutral-70;
            background-color: $neutral-6-5;
            border-radius: 2px;
        }
    }

    .action-button-disabled {
        font-size: 14px;
        font-weight: 600;
        height: 32px;
        border: 1px solid $neutral-4;
        background-color: $neutral-4;
        color: $neutral-10;
        pointer-events: none;
        border-radius: 2px;
        border-width: 10px;
    }

    .action-button-enabled {
        font-size: 14px;
        font-weight: 600;
        height: 32px;
        border: 1px solid $communication-primary;
        background-color: $communication-primary;
        color: $always-white;
        border-radius: 2px;
        border-width: 10px;

        &:hover {
            background-color: $communication-primary;
            color: $always-white;
            border-radius: 2px;
            border-width: 10px;
        }
    }
}
