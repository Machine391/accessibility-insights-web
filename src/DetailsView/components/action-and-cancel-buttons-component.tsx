// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { isEmpty } from 'lodash';
import * as React from 'react';
import styles from './action-and-cancel-buttons-component.scss';

export interface ActionAndCancelButtonsComponentProps {
    isHidden: boolean;
    primaryButtonDisabled: boolean;
    primaryButtonText: string;
    primaryButtonOnClick: (ev) => void;
    cancelButtonOnClick: (ev) => void;
    primaryButtonHref?: string;
    primaryButtonDataAutomationId?: string;
}

export class ActionAndCancelButtonsComponent extends React.Component<ActionAndCancelButtonsComponentProps> {
    public render(): JSX.Element {
        return (
            <div className={styles.actionAndCancelButtonsComponent} hidden={this.props.isHidden}>
                <div className={styles.actionCancelButtonCol}>
                    <Button
                        className={styles.cancelButton}
                        onClick={this.props.cancelButtonOnClick}
                    >
                        Cancel
                    </Button>
                </div>
                <div className={styles.actionCancelButtonCol}>
                    <Button
                        appearance="primary"
                        className={
                            this.props.primaryButtonDisabled
                                ? styles.actionButtonDisabled
                                : styles.actionButtonEnabled
                        }
                        data-automation-id={this.props.primaryButtonDataAutomationId}
                        onClick={this.props.primaryButtonOnClick}
                        disabled={this.props.primaryButtonDisabled}
                        href={this.props.primaryButtonHref}
                        target={isEmpty(this.props.primaryButtonHref) ? '_self' : '_blank'}
                    >
                        {' '}
                        {this.props.primaryButtonText}{' '}
                    </Button>
                </div>
            </div>
        );
    }
}
