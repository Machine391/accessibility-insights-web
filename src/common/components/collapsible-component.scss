// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    display: inline-flex;
    font-size: 26px;
    height: auto;
    width: auto;
    text-align: left;
    color: #106ebe;

    h2 {
        font-weight: normal;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
