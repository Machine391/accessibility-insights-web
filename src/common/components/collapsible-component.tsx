// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ActionButton } from '@fluentui/react';
import { Button, themeToTokensObject, webLightTheme } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import * as React from 'react';
import { ChevronDown24Regular, ChevronRight24Regular } from '@fluentui/react-icons';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    private readonly iconNameDown = 'ChevronDown';
    private readonly iconNameUp = 'ChevronRight';

    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let iconName = this.iconNameUp;
        let content: JSX.Element | null = null;

        if (showContent) {
            iconName = this.iconNameDown;
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                {/* 111<ActionButton
                    className={styles.collapsible}
                    iconProps={{ iconName: iconName }}
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {this.props.header}
                </ActionButton> */}
                <Button className={styles.collapsible} appearance='transparent'
                    onClick={this.onClick}
                    aria-expanded={showContent}>
                    {showContent ? (
                        <ChevronDown24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    ) : (
                        <ChevronRight24Regular color={themeToTokensObject(webLightTheme).colorCompoundBrandStrokeHover} className={styles.collapsible} />
                    )}
                    111
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
