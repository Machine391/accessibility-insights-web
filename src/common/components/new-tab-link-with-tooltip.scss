// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }

    &:focus-visible {
        box-shadow: $neutral-60 0 0 0 1px inset !important;
        outline: $neutral-60 auto 1px !important;
        text-decoration: none !important;
    }
}
