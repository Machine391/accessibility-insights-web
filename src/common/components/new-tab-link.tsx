// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Link, LinkProps, mergeClasses } from '@fluentui/react-components';
import * as React from 'react';

import { NamedFC } from '../react/named-fc';

export type NewTabLinkProps = LinkProps & { as?: 'a' };

export const NewTabLink = NamedFC<NewTabLinkProps>('NewTabLink', ({ className, ...props }) => {
    return <Link className={mergeClasses('insights-link', className)} target="_blank" {...props} />;
});
