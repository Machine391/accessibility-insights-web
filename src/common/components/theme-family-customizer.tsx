// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ThemeProvider, PartialTheme } from '@fluentui/react';
import { FluentProvider, webDarkTheme, webLightTheme } from '@fluentui/react-components';
import { NamedFC } from 'common/react/named-fc';
import { DefaultTheme } from 'common/styles/default-theme';
import { FastPassTheme } from 'common/styles/fast-pass-theme';
import { FastPassV9Theme } from 'common/styles/fast-pass-v9-theme';
import { HighContrastTheme } from 'common/styles/high-contrast-theme';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import * as React from 'react';

export type ThemeFamily = 'default' | 'fast-pass';

const themeFamilyDefaultThemes: { [themeFamily in ThemeFamily]: PartialTheme } = {
    default: DefaultTheme,
    'fast-pass': FastPassTheme,
};

const themeFamilyDefaultV9Themes: { [themeFamily in ThemeFamily] } = {
    default: webLightTheme,
    'fast-pass': FastPassV9Theme,
};

export type ThemeFamilyCustomizerProps = {
    userConfigurationStoreData: UserConfigurationStoreData;
    themeFamily: ThemeFamily;
    children: React.ReactNode;
};

// This is a store-state-aware version of the Fluent UI <ThemeProvider> component. It's intended
// as a wrapper for Fluent UI-based components that need to use a different Theme from our default
// one, but still want to support being overriden by HighContrastTheme when a user has enabled our
// High Contrast Mode setting (or a native system High Contrast setting).
export const ThemeFamilyCustomizer = NamedFC<ThemeFamilyCustomizerProps>(
    'ThemeCustomizer',
    props => {
        const isHighContrastEnabled = props.userConfigurationStoreData?.enableHighContrast === true;
        const defaultTheme = themeFamilyDefaultThemes[props.themeFamily];
        const defaultV9Theme = themeFamilyDefaultV9Themes[props.themeFamily]
        const activeTheme = isHighContrastEnabled ? HighContrastTheme : defaultTheme;
        const activeV9Theme = isHighContrastEnabled ? { ...webDarkTheme, colorNeutralBackground1: '#161616 ' } : defaultV9Theme;
        return <ThemeProvider applyTo="body" theme={activeTheme}><FluentProvider theme={activeV9Theme}>{props.children}</FluentProvider></ThemeProvider>;
    },
);
