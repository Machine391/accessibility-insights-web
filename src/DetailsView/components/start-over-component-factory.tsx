// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { IButton, IContextualMenuItem, IRefObject } from '@fluentui/react';
import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { VisualizationStoreData } from 'common/types/store-data/visualization-store-data';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import { DetailsRightPanelConfiguration } from 'DetailsView/components/details-view-right-panel';
import { StartOverDialogType } from 'DetailsView/components/start-over-dialog';
import {
    DropdownDirection,
    StartOverDropdown,
    StartOverProps,
} from 'DetailsView/components/start-over-dropdown';
import * as React from 'react';
import styles from './start-over-menu-item.scss';

export type StartOverFactoryDeps = {
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
    getProvider: () => AssessmentsProvider;
};

export type StartOverFactoryProps = {
    deps: StartOverFactoryDeps;
    assessmentStoreData: AssessmentStoreData;
    rightPanelConfiguration: DetailsRightPanelConfiguration;
    visualizationStoreData: VisualizationStoreData;
    openDialog: (dialogType: StartOverDialogType) => void;
    buttonRef: IRefObject<IButton>;
    withComponent: boolean
};

export type StartOverMenuItem = Omit<IContextualMenuItem, 'key'>;

export interface StartOverComponentFactory {
    getStartOverComponent: (props: StartOverFactoryProps) => JSX.Element;
    getStartOverMenuItem: (props: StartOverFactoryProps) => StartOverMenuItem;
}

export const AssessmentStartOverFactory: StartOverComponentFactory = {
    getStartOverComponent: props => getStartOverComponentForAssessment(props, 'down'),
    getStartOverMenuItem: props => {
        return {
            // onRender: () => (
            //     <div role="menuitem">{getStartOverComponentForAssessment(props, 'left')}</div>
            // ),
            children: <div role="menuitem">{getStartOverComponentForAssessment(props, 'left')}</div>
        };
    },
};

export const QuickAssessStartOverFactory: StartOverComponentFactory = {
    getStartOverComponent: props => getStartOverComponentForQuickAssess(props, 'down'),
    getStartOverMenuItem: props => {
        return {
            // onRender: () => (
            //     <div role="menuitem">{getStartOverComponentForQuickAssess(props, 'left')}</div>
            // ),
            children: <div role="menuitem">{getStartOverComponentForQuickAssess(props, 'left')}</div>,
        };
    },
};

export const FastpassStartOverFactory: StartOverComponentFactory = {
    getStartOverComponent: props => {
        return (
            <>
                <h1>asdasdasd</h1>
                <InsightsCommandButton {...getStartOverPropsForFastPass(props)} />
            </>
        );
    },
    getStartOverMenuItem: props => getStartOverPropsForFastPassForMenu(props),
};

export function getStartOverComponentForAssessment(
    props: StartOverFactoryProps,
    dropdownDirection: DropdownDirection,
): JSX.Element {
    const selectedTest = props.assessmentStoreData.assessmentNavState.selectedTestType;
    const test = props.deps.getProvider().forType(selectedTest);
    const startOverProps: StartOverProps = {
        singleTestSuffix: test!.title,
        allTestSuffix: 'Assessment',
        dropdownDirection,
        openDialog: props.openDialog,
        buttonRef: props.buttonRef,
        rightPanelOptions: props.rightPanelConfiguration.startOverContextMenuKeyOptions,
        switcherStartOverPreferences: { showTest: true },
    };

    return <StartOverDropdown {...startOverProps} />;
}

export function getStartOverComponentForQuickAssess(
    props: StartOverFactoryProps,
    dropdownDirection: DropdownDirection,
): JSX.Element {
    // since we do not show start over per requirement in quick assess.
    const showTest = false;
    const singleTestSuffix = '';

    const startOverProps: StartOverProps = {
        singleTestSuffix,
        allTestSuffix: 'Quick Assess',
        dropdownDirection,
        openDialog: props.openDialog,
        buttonRef: props.buttonRef,
        rightPanelOptions: props.rightPanelConfiguration.startOverContextMenuKeyOptions,
        switcherStartOverPreferences: { showTest },
    };

    return <StartOverDropdown {...startOverProps} />;
}

export const startOverAutomationId = 'start-over';

export function getStartOverPropsForFastPass(props: StartOverFactoryProps): StartOverMenuItem {
    const selectedTest = props.visualizationStoreData.selectedFastPassDetailsView;
    const detailsViewActionMessageCreator = props.deps.detailsViewActionMessageCreator;
    console.warn("inside getStartOverPropsForFastPass props-->", props);
    const startProps = {
        iconProps: { className: styles.startOverMenuItemIcon },
        iconName: 'ArrowClockwiseRegular',
        onClick: event => detailsViewActionMessageCreator.rescanVisualization(selectedTest, event),
        disabled: props.visualizationStoreData.scanning !== null,
        'data-automation-id': startOverAutomationId,
        text: 'Start over2222',
        className: styles.startOverMenuItem,
    }


    return { ...startProps }
}

export function getStartOverPropsForFastPassForMenu(props: StartOverFactoryProps): StartOverMenuItem {
    const selectedTest = props.visualizationStoreData.selectedFastPassDetailsView;
    const detailsViewActionMessageCreator = props.deps.detailsViewActionMessageCreator;
    const startProps = {
        iconProps: { className: styles.startOverMenuItemIcon },
        iconName: 'ArrowClockwiseRegular',
        onClick: event => detailsViewActionMessageCreator.rescanVisualization(selectedTest, event),
        disabled: props.visualizationStoreData.scanning !== null,
        'data-automation-id': startOverAutomationId,
        text: 'Start over',
        className: styles.startOverMenuItem,
    }

    // if (props.withComponent) {
    //     return { children: <InsightsCommandButton {...startProps} /> }
    // }
    return { children: <InsightsCommandButton {...startProps} /> }
    //return { ...startProps }
    // return {
    //     iconProps: { className: styles.startOverMenuItemIcon },
    //     iconName: 'ArrowClockwiseRegular',
    //     onClick: event => detailsViewActionMessageCreator.rescanVisualization(selectedTest, event),
    //     disabled: props.visualizationStoreData.scanning !== null,
    //     'data-automation-id': startOverAutomationId,
    //     text: 'Start over2222',
    //     className: styles.startOverMenuItem,
    // };
    //     return {
    //         withoutComponent === true ? ...startProps: children: <InsightsCommandButton {...startProps} />
    // }
}
