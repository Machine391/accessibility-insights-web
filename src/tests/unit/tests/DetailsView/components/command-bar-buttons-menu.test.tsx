// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { CommandBarButton, IButton, IOverflowSetItemProps, RefObject } from '@fluentui/react';
import {
    Menu,
    MenuButton,
    MenuItem,
    MenuList,
    MenuPopover,
    MenuTrigger,
    Tooltip,
} from '@fluentui/react-components';
import { fireEvent, render, screen } from '@testing-library/react';
import {
    CommandBarButtonsMenu,
    CommandBarButtonsMenuProps,
} from 'DetailsView/components/command-bar-buttons-menu';
import { StartOverMenuItem } from 'DetailsView/components/start-over-component-factory';
import * as React from 'react';
import {
    expectMockedComponentPropsToMatchSnapshots,
    getMockComponentClassPropsForCall,
    mockReactComponents,
    useOriginalReactElements,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { IMock, Mock, Times } from 'typemoq';

jest.mock('@fluentui/react-components');
jest.mock('common/icons/fluentui-v9-icons');

describe('CommandBarButtonsMenu', () => {
    mockReactComponents([Tooltip, Menu, MenuTrigger, MenuButton, MenuPopover, MenuList, MenuItem]);

    let renderExportReportComponentMock: IMock<() => JSX.Element>;
    let getStartOverMenuItemMock: any;
    let commandBarButtonsMenuProps: CommandBarButtonsMenuProps;

    beforeEach(() => {
        renderExportReportComponentMock = Mock.ofInstance(() => null);
        getStartOverMenuItemMock = Mock.ofInstance(() => null);
        commandBarButtonsMenuProps = {
            renderExportReportButton: renderExportReportComponentMock.object,
            saveAssessmentButton: <>Save assessment button</>,
            loadAssessmentButton: <>Load assessment button</>,
            transferToAssessmentButton: <>Transfer to assessment button</>,
            getStartOverMenuItem: getStartOverMenuItemMock.object,
            buttonRef: {
                current: {},
            },
            hasSubMenu: false,
        };
    });

    it('renders CommandBarButtonsMenu', () => {
        const wrapper = render(<CommandBarButtonsMenu {...commandBarButtonsMenuProps} />);
        //expectMockedComponentPropsToMatchSnapshots([Menu]);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });

    it('renders all child buttons with hasSubMenu false,', () => {
        //useOriginalReactElements('@fluentui/react', ['TooltipHost']);
        useOriginalReactElements('@fluentui/react-components', [
            'Menu',
            'MenuButton',
            'MenuTrigger',
            'MenuPopover',
            'MenuList',
            'MenuItem',
            'Tooltip',
        ]);
        setupExportReportMenuItem();
        setupStartOverMenuItem();

        const test = render(<CommandBarButtonsMenu {...commandBarButtonsMenuProps} />);
        test.debug();
        const getMoreActionsButton = test.queryByRole('button');
        fireEvent.click(getMoreActionsButton);
        //test.debug();
        const hasExportButton = screen.getByText('Report export button');
        const hasSaveButton = screen.getByText('Save assessment button');
        const hasLoadButton = screen.getByText('Load assessment button');
        //const hasStartOver = screen.getByText('Start over button');
        const hasTransferButton = screen.getByText('Transfer to assessment button');
        expect(hasExportButton).toMatchSnapshot('render export report menuitem');
        expect(hasSaveButton).toMatchSnapshot('render save assessment menuitem');
        expect(hasLoadButton).toMatchSnapshot('render load assessment menuitem');
        expect(hasTransferButton).toMatchSnapshot('render transfer to assessment menuitem');
        // expect(hasStartOver).toMatchSnapshot('render start over button menuitem');
        //old
        //expect(overflowItems[4].onRender()).toMatchSnapshot('render start over menuitem');
        // expect(menuListValues[0])
        //menuListValues.debug();

        // expect(menuListValues).toMatchSnapshot('render export report menuitem')

        //const commandBarProps = getMockComponentClassPropsForCall(CommandBarButton);

        // const overflowItems: IOverflowSetItemProps[] = commandBarProps?.menuProps?.items;

        // expect(commandBarProps?.menuProps?.items).toBeDefined();
        // expect(commandBarProps?.menuProps?.items).toHaveLength(5);

        // expect(overflowItems[0].onRender()).toMatchSnapshot('render export report menuitem');
        // expect(overflowItems[1].onRender()).toMatchSnapshot('render save assessment menuitem');
        // expect(overflowItems[2].onRender()).toMatchSnapshot('render load assessment menuitem');
        // expect(overflowItems[3].onRender()).toMatchSnapshot(
        //     'render transfer to assessment menuitem',
        // );
        // expect(overflowItems[4].onRender()).toMatchSnapshot('render start over menuitem');

        renderExportReportComponentMock.verifyAll();
        getStartOverMenuItemMock.verifyAll();
    });


    function setupExportReportMenuItem(): void {
        renderExportReportComponentMock
            .setup(r => r())
            .returns(() => <>Report export button</>)
            .verifiable(Times.once());
    }

    function setupStartOverMenuItem(): void {
        const startOverMenuItem = {
            children: <>Start over button</>,
        };
        getStartOverMenuItemMock
            .setup(s => s())
            .returns(() => startOverMenuItem.children)
            .verifiable(Times.once());
    }
});
