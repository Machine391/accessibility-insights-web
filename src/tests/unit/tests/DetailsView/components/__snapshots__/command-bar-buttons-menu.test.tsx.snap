// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandBarButtonsMenu renders CommandBarButtonsMenu 1`] = `
<DocumentFragment>
  <mock-styledtooltiphostbase
    classname="commandBarButtonsMenu"
    content="More actions"
  >
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="fui-Button r1alrhcs fui-MenuButton commandBarButtonsMenuButton ___1j82qye_10z14xp f18ktai2 fwbmr0d f44c6la"
      data-tabster="{"restorer":{"type":1}}"
      id="menu1"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2 fui-MenuButton__icon"
      >
        <svg
          aria-hidden="true"
          class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.25 10a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0Zm5 0a1.25 1.25 0 1 1-2.5 0 1.25 1.25 0 0 1 2.5 0ZM15 11.25a1.25 1.25 0 1 0 0-2.5 1.25 1.25 0 0 0 0 2.5Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
  </mock-styledtooltiphostbase>
</DocumentFragment>
`;
